test/proxy/BeaconProxy.test.js

```diff
@@ -0,0 +1,493 @@
+        await beaconContract.upgradeTo(instance2.address, { from: proxyCreator });
+
+        await expectRevert.unspecified(
+          sendTransaction(proxy, '', [], [], { from: anotherAccount }),
+        );
+
+        const proxyInstance2 = await new Implementation2(proxy.address);
+        const res = await proxyInstance2.getValue();
+        expect(res).to.bignumber.eq('0');
+      });
+    });
+  });
+
+  describe('one beacon for many proxies', function () {
+    describe('initialize', function () {
+      const proxy1ExpectedInitializedValue = '10';
+      const proxy2ExpectedInitializedValue = '42';
+
+      beforeEach(async function () {
+        this.beacon = await UpgradeableBeacon.new(this.implementationV0.address, { from: proxyCreator });
+        const proxy1InitializeData = this.implementationV0.contract.methods
+          .initializeNonPayableWithValue(proxy1ExpectedInitializedValue)
+          .encodeABI();
+        this.proxy1 = await BeaconProxy.new(this.beacon.address, proxy1InitializeData, { from: proxyCreator });
+        const proxy2InitializeData = this.implementationV0.contract.methods
+          .initializeNonPayableWithValue(proxy2ExpectedInitializedValue)
+          .encodeABI();
+        this.proxy2 = await BeaconProxy.new(this.beacon.address, proxy2InitializeData, { from: proxyCreator });
+      });
+
+      it('initializes the proxy1', async function () {
+        const dummy = await new DummyImplementation(this.proxy1.address);
+        expect(await dummy.value()).to.bignumber.eq(proxy1ExpectedInitializedValue);
+      });
+
+      it('initializes the proxy2 with a different value', async function () {
+        const dummy = await new DummyImplementation(this.proxy2.address);
+        expect(await dummy.value()).to.bignumber.eq(proxy2ExpectedInitializedValue);
+      });
+    });
+
+    describe('upgrade', function () {
+      beforeEach(async function () {
+        const instance2 = await Implementation2.new();
+        this.beacon = await UpgradeableBeacon.new(instance2.address, { from: proxyCreator });
+        const initializeData = Buffer.from('');
+        this.proxy1 = await BeaconProxy.new(this.beacon.address, initializeData, { from: proxyCreator });
+        this.proxy2 = await BeaconProxy.new(this.beacon.address, initializeData, { from: proxyCreator });
+
+        this.instance3 = await Implementation3.new();
+        const beacon = new UpgradeableBeacon(this.beacon.address);
+        await beacon.upgradeTo(this.instance3.address, { from: proxyCreator });
+      });
+
+      it('UpgradeableBeacon has the correct implementation', async function () {
+        const implementation = await this.beacon.implementation();
+        expect(implementation).to.be.equal(this.instance3.address);
+      });
+
+      it('should remove function from proxy 1', async function () {
+        const instance2 = await new Implementation2(this.proxy1.address);
+        await expectRevert.unspecified(instance2.getValue());
```

Comment:
Let's add a test for a successful operation. I don't feel confident that an upgrade succeeded if the only thing we're testing is an unspecified revert.